--- a/assets/survival/shaders/mechpower.fsh
+++ b/assets/survival/shaders/mechpower.fsh
@@ -8,6 +8,7 @@
 in float glowLevel;
 flat in int renderFlags;
 in vec3 normal;
+in vec3 vertPos;
 in float normalShadeIntensity;
 
 layout(location = 0) out vec4 outColor;
@@ -33,7 +34,8 @@
   
   //outColor = vec4((normal.x + 0.5) / 2, (normal.y + 0.5)/2, (normal.z+0.5)/2, 1);
 
-  outGlow = vec4(glowLevel, 0, 0, outColor.a);
+  float scatterAmt = calculateVolumetricScatter(vertPos);
+  outGlow = vec4(glowLevel, scatterAmt, 0, outColor.a);
   
 #if SSAOLEVEL > 0
 	outGPosition = vec4(fragPosition.xyz, fogAmount + glowLevel);
