--- orig/assets/game/shaderincludes/fogandlight.vsh
+++ patched/assets/game/shaderincludes/fogandlight.vsh
@@ -1,4 +1,4 @@
-#ifndef DYNLIGHTS
+﻿#ifndef DYNLIGHTS
     #define DYNLIGHTS 0
 #endif
 #if SHADOWQUALITY > 0
@@ -84,7 +84,7 @@
 	}
 
 	// Light up all caves
-	// bBlock = 1;
+	//bBlock = 1;
 
 	if (bPointBrightSum > 0) {
 		pointColSum.rgb /= max(1, bPointBrightSum);
@@ -143,7 +143,6 @@
 }
 
 
-
 vec4 applyFog(vec4 worldPos, vec4 rgbaPixel, vec4 rgbaFog, float fogMin, float fogDensity) {
 	float amount = getFogLevel(worldPos, fogMin, fogDensity);
 	return vec4(mix(rgbaPixel.rgb, rgbaFog.rgb, amount), rgbaPixel.a * rgbaFog.a);
