--- a/assets/game/shaderincludes/fogandlight.vsh
+++ b/assets/game/shaderincludes/fogandlight.vsh
@@ -1,4 +1,4 @@
-#ifndef DYNLIGHTS
+﻿#ifndef DYNLIGHTS
     #define DYNLIGHTS 0
 #endif
 #if SHADOWQUALITY > 0
@@ -84,7 +84,7 @@
 	}
 
 	// Light up all caves
-	// bBlock = 1;
+	//bBlock = 1;
 
 	if (bPointBrightSum > 0) {
 		pointColSum.rgb /= max(1, bPointBrightSum);
@@ -137,13 +137,12 @@
 	val = max(min(0.04, val), val - nearnessToPlayer);
 	
 	// Needs to be added after so that underwater fog still gets applied. 
-	val += fogMin; 
+	val += fogMin;
 	
 	return clamp(val, 0, 1);
 }
 
 
-
 vec4 applyFog(vec4 worldPos, vec4 rgbaPixel, vec4 rgbaFog, float fogMin, float fogDensity) {
 	float amount = getFogLevel(worldPos, fogMin, fogDensity);
 	return vec4(mix(rgbaPixel.rgb, rgbaFog.rgb, amount), rgbaPixel.a * rgbaFog.a);
