--- a/assets/game/shaders/blur.vsh
+++ b/assets/game/shaders/blur.vsh
@@ -16,14 +16,14 @@
 		float pixelSize = 1.0 / frameSize.y;
 		
 		for (int i = -8; i < 8; i++) {
-			texCoords[i + 8] = texCoord + vec2(0, pixelSize * i);
+			texCoords[i + 8] = texCoord + vec2(0, pixelSize * i * 4);
 		}
 
 	} else {
 		float pixelSize = 1.0 / frameSize.x;
 		
 		for (int i = -8; i < 8; i++) {
-			texCoords[i + 8] = texCoord + vec2(pixelSize * i, 0);
+			texCoords[i + 8] = texCoord + vec2(pixelSize * i * 4, 0);
 		}
 	}
 }
\ No newline at end of file
