--- orig/assets/game/shaders/celestialobject.fsh
+++ patched/assets/game/shaders/celestialobject.fsh
@@ -1,4 +1,4 @@
-#version 330 core
+﻿#version 330 core
 
 in vec2 uv;
 in vec4 color;
@@ -33,7 +33,7 @@
 void main () {
 	vec4 texColor = applyFog(texture(tex, uv) * color, 0);
 	if (texColor.a < alphaTest) discard;
-	vec4 texGlow = vec4(glowLevel, extraGodray, 0, texColor.a);
+	vec4 texGlow = vec4(glowLevel, dayLight*0.2f, 0, texColor.a);
 	
 	vec4 skyColor = vec4(1);
 	vec4 skyGlow = vec4(1);
