--- a/assets/game/shaders/chunktransparent.vsh
+++ b/assets/game/shaders/chunktransparent.vsh
@@ -1,4 +1,4 @@
-#version 330 core
+﻿#version 330 core
 #extension GL_ARB_explicit_attrib_location: enable
 
 layout(location = 0) in vec3 vertexPositionIn;
@@ -16,6 +16,9 @@
 layout(location = 4) in int colormapData;
 
 
+#if GODRAYS > 0
+uniform vec3 lightPosition;
+#endif
 
 uniform vec4 rgbaFogIn;
 uniform vec3 rgbaAmbientIn;
@@ -74,4 +77,8 @@
 	
 	normal = unpackNormal(renderFlags >> 7);
 	normalShadeIntensity = min(1, rgbaLightIn.a * 1.5);
+
+#if GODRAYS > 0
+	prepareVolumetricLighting(lightPosition);
+#endif
 }
\ No newline at end of file
