--- orig/assets/game/shaders/chunkliquid.vsh
+++ patched/assets/game/shaders/chunkliquid.vsh
@@ -1,4 +1,4 @@
-#version 330 core
+﻿#version 330 core
 #extension GL_ARB_explicit_attrib_location: enable
 
 layout(location = 0) in vec3 vertexPositionIn;
@@ -46,6 +46,10 @@
 uniform vec3 playerViewVec;
 uniform vec3 sunPosRel;
 
+#if GODRAYS > 0
+uniform vec3 lightPosition;
+#endif
+
 out vec2 flowVectorf;
 out vec4 rgba;
 out vec4 rgbaFog;
@@ -87,11 +91,6 @@
 		waterStillCounterOff = 1;
 	}
 	
-	
-	
-	
-	
-	
 	fragWorldPos = worldPos.xyz + playerpos;
 	fWorldPos = worldPos.xyz;
 	
@@ -132,6 +131,9 @@
 	calcShadowMapCoords(modelViewMatrix, worldPos);
 	calcColorMapUvs(colormapData, vec4(vertexPositionIn + origin, 1.0) + vec4(playerpos, 1), rgbaLightIn.a, false);
 	
+#if GODRAYS > 0
+	prepareVolumetricLighting(lightPosition);
+#endif
 	
 	// We pretend the decal is closer to the camera to enforce it always being drawn on top
 	// Required e.g. when water is besides stairs or slabs
