--- a/assets/game/shaders/chunktransparent.fsh
+++ b/assets/game/shaders/chunktransparent.fsh
@@ -1,7 +1,8 @@
-#version 330 core
+﻿#version 330 core
 #extension GL_ARB_explicit_attrib_location: enable
 
 uniform sampler2D terrainTex;
+uniform float fogDensityIn;
 
 in vec4 rgba;
 in vec4 rgbaFog;
@@ -27,6 +28,8 @@
 void drawPixel(vec4 color) {
 	float weight = color.a * clamp(0.03 / (1e-5 + pow(gl_FragCoord.z / 200, 4.0)), 1e-2, 3e3);
 	
+	float scatterAmt = calculateVolumetricScatter(worldPos.xyz);
+
     // RGBA32F texture (accumulation)
     outAccu = vec4(color.rgb * color.a, color.a) * weight;
 	
@@ -34,7 +37,7 @@
     // Make sure to use the red channel (and GL_RED target in your texture)
     outReveal.r = color.a;
 
-    outGlow = vec4(glowLevel, 0, 0, color.a);
+    outGlow = vec4(glowLevel, scatterAmt, 0, color.a);
 }
 
 
@@ -46,9 +49,9 @@
 
 	vec4 texColor = texture(terrainTex, uv) * rgba * getColorMapping(terrainTex);
 
-	texColor = applyFogAndShadowWithNormal(texColor, fogAmount, normal, normalShadeIntensity, 0.45);
+	texColor = applyOverexposedFogAndShadow(texColor, fogAmount, normal, normalShadeIntensity, 0.45, worldPos.xyz, fogDensityIn);
 
-#if SHINYEFFECT > 0
+#if SHINYEFFECT > 0 && VSMOD_SSR == 0
 	// Shiny bit flag
 	if (((renderFlags >> 5) & 1) > 0) {
 		vec3 worldVec = normalize(worldPos.xyz);
