--- a/assets/game/shaders/clouds.vsh
+++ b/assets/game/shaders/clouds.vsh
@@ -1,4 +1,4 @@
-#version 330 core
+﻿#version 330 core
 #extension GL_ARB_explicit_attrib_location: enable
 
 // Per vertex attribute
@@ -34,6 +34,10 @@
 uniform float cloudYTranslate; // Already added to the modelViewMatrix
 uniform vec2 tileOffset;
 
+#if GODRAYS > 0
+uniform vec3 lightPosition;
+#endif
+
 
 uniform mat4 projectionMatrix;
 uniform mat4 modelViewMatrix;
@@ -138,4 +142,10 @@
 	}
 	
 	rgbaCloud.a = clamp(rgbaCloud.a, 0, 1);
+	
+#if GODRAYS > 0
+	prepareVolumetricLighting(lightPosition);
+#endif
+
+	calcShadowMapCoords(modelViewMatrix, vec4(vexPos, 1.0));
 }
