--- a/assets/game/shaders/entityanimated.vsh
+++ b/assets/game/shaders/entityanimated.vsh
@@ -1,4 +1,4 @@
-#version 330 core
+﻿#version 330 core
 #extension GL_ARB_explicit_attrib_location: enable
 
 layout(location = 0) in vec3 vertexPositionIn;
@@ -15,6 +15,10 @@
 uniform vec4 renderColor;
 uniform int addRenderFlags;
 
+#if GODRAYS > 0
+uniform vec3 lightPosition;
+#endif
+
 uniform mat4 projectionMatrix;
 uniform mat4 viewMatrix;
 uniform mat4 modelMatrix;
@@ -28,6 +32,7 @@
 out vec4 color;
 out vec4 rgbaFog;
 out float fogAmount;
+out vec3 vertexPosition;
 
 flat out int renderFlags;
 
@@ -49,6 +54,7 @@
 	vec4 worldPos = animModelMat * vec4(vertexPositionIn, 1.0);
 	
 	worldPos = applyVertexWarping(flags | addRenderFlags, worldPos);
+	vertexPosition = worldPos.xyz;
 	
 	// Cheap fix to "not render" head in first person mode
 	if (jointId == skipRenderJointId || jointId == skipRenderJointId2) {
@@ -77,4 +83,8 @@
 		fragPosition = cameraPos;
 		gnormal = viewMatrix * vec4(normal, 0);
 	#endif
+
+	#if GODRAYS > 0
+	prepareVolumetricLighting(lightPosition);
+	#endif
 }
\ No newline at end of file
