--- a/assets/game/shaders/standard.fsh
+++ b/assets/game/shaders/standard.fsh
@@ -14,6 +14,8 @@
 uniform float extraGodray = 0;
 uniform float alphaTest = 0.001;
 uniform float ssaoAttn = 0;
+uniform float extraOutGlow = 0;
+uniform float fogDensityIn;
 
 
 // Texture overlay "hack"
@@ -34,6 +36,7 @@
 in vec4 rgbaGlow;
 flat in int renderFlags;
 in vec3 normal;
+in vec4 vertexPosition;
 
 
 
@@ -72,7 +75,7 @@
 //	outColor.rgb += rgbaGlow.rgb * min(0.8, glowLevel * rgbaGlow.a);
 	//outColor.rgb = rgbaGlow.rgb;
 	
-	outColor = applyFogAndShadow(outColor, fogAmount);
+	outColor = applyOverexposedFogAndShadowFlat(outColor, fogAmount, normal, vertexPosition.xyz, fogDensityIn);
 	//outColor=vec4(1,0,0,1);
 	
 	if (outColor.a < alphaTest) discard;
@@ -82,6 +85,7 @@
 #if SHINYEFFECT > 0	
 	glow = pow(max(0, dot(normal, lightPosition)), 6) / 8 * shadowIntensity * (1 - fogAmount);
 #endif
+	glow += extraOutGlow;
 
 #if SSAOLEVEL > 0
 	outGPosition = vec4(fragPosition.xyz, fogAmount + glowLevel);
